demo-projects/meetup/site/pages/signup.js

```diff
@@ -48,48 +55,76 @@ export default () => {
         <title>Join | {meetup.name}</title>
       </Head>
       <Navbar background="white" foreground={colors.greyDark} />
-      <Container css={{ marginTop: gridSize * 3 }}>
-        <H1>Join</H1>
-
-        {errorState && (
-          <p css={{ color: colors.red }}>Please check your email and password then try again.</p>
-        )}
+      <Mutation
+        mutation={CREATE_USER}
+        onCompleted={() => {
+          handleSignin();
+        }}
+      >
+        {(createUser, { error: mutationError }) => {
+          return (
+            <Container css={{ marginTop: gridSize * 3 }}>
+              <H1>Join</H1>
+              {mutationError && (
+                <p css={{ color: colors.red }}>The email provided is already in use.</p>
+              )}
+              {errorState && <p css={{ color: colors.red }}>An unknown error has occured</p>}
 
-        <form css={{ marginTop: gridSize * 3 }} noValidate onSubmit={handleSubmit}>
-          <Field>
-            <Label htmlFor="email">Email</Label>
-            <Input
-              required
-              type="text"
-              autoFocus
-              autoComplete="email"
-              placeholder="you@awesome.com"
-              disabled={isLoading || isAuthenticated}
-              value={email}
-              onChange={e => setEmail(e.target.value)}
-            />
-          </Field>
-          <Field>
-            <Label htmlFor="password">Password</Label>
-            <Input
-              required
-              type="password"
-              id="password"
-              minLength="8"
-              placeholder="supersecret"
-              autoComplete="password"
-              disabled={isLoading || isAuthenticated}
-              value={password}
-              onChange={e => setPassword(e.target.value)}
-            />
-          </Field>
-          {isLoading ? (
-            <Button disabled>Creating account...</Button>
-          ) : (
-            <Button type="submit">Sign up</Button>
-          )}
-        </form>
-      </Container>
+              <form
+                css={{ marginTop: gridSize * 3 }}
+                noValidate
+                onSubmit={() => handleSubmit(createUser)}
```

Comment:
Really should be passing the event through from here, so we can cancel default correctly (unless I'm mistaken how this api works)